### PR TITLE
📊 minerals: Visualize IEA critical minerals

### PR DESCRIPTION
**Pull request by pabloarosado**
* Improve IEA's critical minerals dataset.
* Create charts for mineral demand for different clean technologies.
* Create a new explorer for supply and demand prospects (there are still some open issues with a few minerals).

### CODE REVIEW

File: etl/steps/export/explorers/minerals/latest/minerals_supply_and_demand_prospects.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        & (df_graphers["Demand Scenario Dropdown"] == "Net zero by 2050")
+        & (df_graphers["Metric Radio"] == "Total")
+    )
+    assert len(df_graphers[default_mask]) == 1, "Multiple rows selected for default view."
+    df_graphers.loc[default_mask, "defaultView"] = True
+
+    # Prepare explorer metadata.
+    config = {
+        "explorerTitle": "Minerals Supply and Demand Prospects",
+        "explorerSubtitle": "",
```
Comment (paarriagadap) [inline etl/steps/export/explorers/minerals/latest/minerals_supply_and_demand_prospects.py:148]: I think all explorers have a subtitle in the style of: _Explore data from xxx_, for them to be shown in https://ourworldindata.org/charts

File: etl/steps/export/explorers/minerals/latest/minerals_supply_and_demand_prospects.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        "entityType": "country or technology",
+        # Ensure all entities (countries or technologies) are selected by default, so that stacked area charts are always showing totals.
+        "selection": sorted(set(tb["country"])),
```
Comment (paarriagadap) [inline etl/steps/export/explorers/minerals/latest/minerals_supply_and_demand_prospects.py:151]: You can add `pickerColumnSlugs` if you want to have some columns to sort the entities.

**Review by paarriagadap — APPROVED** [review]
Looks good! I have added some comments below. Messy data, uh?

File: etl/steps/data/garden/iea/2024-07-04/critical_minerals.py [hunk 3]
```diff
@@ -29,13 +29,19 @@
...
-from owid.catalog import Dataset, Table
+from owid.catalog import Dataset, Table, VariablePresentationMeta
 from owid.datautils.dataframes import map_series
+from structlog import get_logger
 
 from etl.data_helpers import geo
 from etl.helpers import PathFinder, create_dataset
 
+# Initialize log.
+log = get_logger()
```
[the remark this answers is not in the document]
Comment (pabloarosado) [reply etl/steps/data/garden/iea/2024-07-04/critical_minerals.py:43]: Nice, I didn't know about this!

File: etl/steps/export/explorers/minerals/latest/minerals_supply_and_demand_prospects.py (hunk 2 above)
Comment (pabloarosado) [reply etl/steps/export/explorers/minerals/latest/minerals_supply_and_demand_prospects.py:151]: I don't understand your suggestion (but I suppose it's not very important).

Comment (paarriagadap) [reply etl/steps/export/explorers/minerals/latest/minerals_supply_and_demand_prospects.py:151]: In the entity selector you can add indicators to sort them, as in here: 
image

**Pull request merged**